spatial_graph/spatial_graph.py

```diff
@@ -16,9 +16,9 @@ def __init__(
         position_attr,
         directed=False,
     ):
-        assert position_attr in node_attr_dtypes, (
-            f"position attribute '{position_attr}' not defined in 'node_attr_dtypes'"
-        )
+        assert (
+            position_attr in node_attr_dtypes
+        ), f"position attribute '{position_attr}' not defined in 'node_attr_dtypes'"
```

Comment:
```suggestion
        if position_attr not in node_attr_dtypes:
            raise ValueError(f"position attribute {position_attr!r} not defined in 'node_attr_dtypes'")
```

(while we're at it... assertion error isn't the right one here)

Reply:
Just for my own reference, when is it appropriate to use an assert versus an if statement/raise error?

Reply:
pretty much only in testing:

- https://docs.astral.sh/ruff/rules/assert/
- https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html#b101-test-for-use-of-assert